File: GPTBackendServer/src/main/kotlin/com/neelesh/model/KnowledgeSpace.kt
package com.neelesh.model

import com.fasterxml.jackson.databind.JsonNode
import org.http4k.contract.openapi.OpenAPIJackson.array
import org.http4k.contract.openapi.OpenAPIJackson.string
import org.http4k.format.Jackson

data class KnowledgeSpace(
    val id: String,
    val name: String,
    val email: String,
    val files: List<String>
) {

    fun toJson() : JsonNode {
        return Jackson.obj(
            "id" to string(id),
            "name" to string(name),
            "email" to string(email),
            "files" to array(files.map { string(it) })
        )
    }

    companion object {

        fun fromJson(jsonNode: JsonNode) : KnowledgeSpace {
            return KnowledgeSpace(
                jsonNode.get("id").textValue(),
                jsonNode.get("name").textValue(),
                jsonNode.get("email").textValue(),
                jsonNode.get("files").map { it.textValue() }
            )
        }

    }

}

File: LLMServer/knowledge_space.py

class KnowledgeSpace:
    def __init__(self, id: str, name: str, files: [str], email: str):
        self.id = id
        self.name = name
        self.files = files
        self.email = email

    def __eq__(self, other):
        if type(other) == KnowledgeFile:
            return other.id == self.id and other.name == self.name and other.files == self.files and self.email == other.email
        return False

    def to_dict(self):
        return {
            "id": self.id,
            "name": self.name,
            "files": self.files,
            "email": self.email
        }

class KnowledgeFile:

    def __init__(self, id: str, email: str, name: str, blobIds: [str], indexDict: str):
        self.id = id
        self.email = email
        self.name = name
        self.blobIds = blobIds
        self.indexDict = indexDict

    def __eq__(self, other):
        if type(other) == KnowledgeFile:
            return other.id == self.email and other.indexDict == self.indexDict and other.name == self.name
        return False

    def to_dict(self):
        return {
            "id": self.id,
            "email": self.email,
            "name": self.name,
            "blobIds": self.blobIds,
            "indexDict": self.indexDict,
        }